src/internal/testutil/cmds.go

```diff
@@ -116,11 +118,64 @@ which match >/dev/null || {
 
 	// do the substitution
 	template.Must(template.New("").Parse(dedent(cmd))).Execute(buf, subsMap)
+	return buf
+}
+
+// BashCmd is a convenience function that:
+// 1. Performs a Go template substitution on 'cmd' using the strings in 'subs'
+// 2. Returns a command that runs the result string from 1 as a Bash script
+func BashCmd(cmd string, subs ...string) *exec.Cmd {
 	res := exec.Command("/bin/bash")
 	res.Stderr = os.Stderr
 	// useful for debugging, but makes logs too noisy:
 	// res.Stdout = os.Stdout
-	res.Stdin = buf
+	res.Stdin = bashCmd(cmd, subs...)
 	res.Env = os.Environ()
 	return res
 }
+
+func PachctlBashCmd(t *testing.T, c *client.APIClient, cmd string, subs ...string) *exec.Cmd {
+	t.Helper()
+
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(bashCmd(cmd, subs...))
+
+	config := fmt.Sprintf("test-pach-config-%s.json", t.Name())
+	if _, err := os.Open(config); err != nil {
+		_, err = os.Create(config)
+		require.NoError(t, err)
+		// remove the empty file so that a config can be generated
+		require.NoError(t, os.Remove(config))
+		t.Cleanup(func() {
+			// since this call gets run multiple times, ignore error
+			_ = os.Remove(config)
+		})
+		return BashCmd(`
+		export PACH_CONFIG={{.config}}
+		pachctl config set context  --overwrite testing <<EOF
```

Comment:
Just wondering: this implies that we can't use `PachctlBashCmd` to test `pachctl config` since it depends on it. Do you think we can move this out somewhere?

Reply:
that's true, I actually decided to go with this approach after looking at the `pachctl config` tests that do something similar. `pachctl config` actually doesn't need an API client to test, so I think it's alright to leave it as is. Those tests are located here: https://github.com/pachyderm/pachyderm/blob/master/src/server/config/cmds_test.go#L13